src/components/KaotoToolbar.tsx

```diff
@@ -102,6 +104,16 @@ export const KaotoToolbar = ({ toggleCatalog, toggleCodeEditor, hideLeftPanel }:
     });
   }, []);
 
+  // configure UI theme
+  useEffect(() => {
+    const uiTheme = localStorage.getItem(LOCAL_STORAGE_UI_THEME_KEY) ?? "true";
+    if (uiTheme === "true") {
+      htmlTagElement?.classList.remove(THEME_DARK_CLASS);
+    } else {
+      htmlTagElement?.classList.add(THEME_DARK_CLASS);
```

Comment:
It feels strange to attach a class to the root document, I'm wondering what would happen if we embed Kaoto editor inside another application, I guess that the dark theme will be applied to the root as well :thinking: . I couldn't find any related docs from https://www.patternfly.org/

Reply:
I know, I haven't find either any info regarding this. Do you think there is a better place to have this logic?

Reply:
I tried to add it to the main div and didn't work. Seems like the [demo react app ](https://github.com/patternfly/patternfly-react/blob/83c54ab517dd329d924d69ade0430c389aaff5b1/packages/react-integration/demo-app-ts/src/App.tsx#L44)actually does it the same way.